src/components/GestureComponents.tsx

```diff
@@ -84,6 +90,31 @@ export const TextInput = createNativeWrapper<RNTextInputProps>(RNTextInput);
 // eslint-disable-next-line @typescript-eslint/no-redeclare
 export type TextInput = typeof TextInput & RNTextInput;
 
+type TextProps = Omit<RNTextProps, 'onPress'> & {
+  onPress?: (
+    event: HandlerStateChangeEvent<NativeViewGestureHandlerPayload>
+  ) => void;
+};
+
+export const Text = (props: TextProps) => {
+  const { children, onPress, ...rest } = props;
+  return (
+    <NativeViewGestureHandler
+      onHandlerStateChange={(event) => {
+        if (
+          event.nativeEvent.oldState === State.ACTIVE &&
+          event.nativeEvent.state === State.END
```

Comment:
Have you tried on iOS? Maybe checking simply the `END` state is enough?

Reply:
I've changed it to new API anyway ([8a0e06f](https://github.com/software-mansion/react-native-gesture-handler/pull/3202/commits/8a0e06ff9543e40ff7150dee65c23db880a9726b))